src/pubsub/ua_pubsub_writergroup.c

```diff
@@ -860,8 +911,17 @@ sendNetworkMessageJson(UA_PubSubConnection *connection, UA_DataSetMessage *dsm,
         goto cleanup;
     UA_assert(bufPos == bufEnd);
 
+    /* Choose the channel */
+    UA_PubSubChannel *channel = NULL;
+
+    if(transportSettings->content.decoded.data) {
+        channel = writerGroup->channel;
+    } else {
+        channel = connection->channel;
+    }
```

Comment:
This can result in a SegFault when the transportSettings->type is DatagramWriterGroupTransportDataType (not DatagramWriterGroupTransport**2**DataType). This can happen when configuring a multicast publisher and setting the transportsettings